user_journey/views/event_type.view.lkml

```diff
@@ -0,0 +1,143 @@
+view: event_type {
+  derived_table: {
+    sql:
+    SELECT
+      mozfun.event_analysis.aggregate_match_strings(
+          ARRAY_AGG(
+            CONCAT(event_properties_match_string,
+                   event_match_string))) AS match_string
+    FROM (
+      SELECT
+        mozfun.event_analysis.event_index_to_match_string(index)
+          AS event_match_string,
+        ARRAY_TO_STRING(ARRAY_AGG(event_property_match_string
+                    ORDER BY
+                      event_property_index DESC), '')
+          AS event_properties_match_string,
+      FROM
+        (
+          SELECT
+            selected_events.index,
+            event_property.index AS event_property_index,
+            COALESCE(event_property_match_string,
+              '.') AS event_property_match_string
+          FROM
+            (
+              SELECT
+                *
+              FROM
+                mozdata.messaging_system.event_types
+              WHERE
+                {% condition message_id %} event_types.category {% endcondition %}
+                AND {% condition event_type %} event_types.event {% endcondition %}
+            ) AS selected_events
+          LEFT JOIN
+            UNNEST(event_properties) AS event_property
+          LEFT JOIN
+            (
+              SELECT
+                selected_events.index,
+                event_property.index AS event_property_index,
+                mozfun.event_analysis.aggregate_match_strings(ARRAY_AGG(mozfun.event_analysis.escape_metachars(event_property_value.value))) AS event_property_match_string
+              FROM
+                (
+                  SELECT
+                    *
+                  FROM
+                    mozdata.messaging_system.event_types
+                  WHERE
+                    {% condition message_id %} event_types.category {% endcondition %}
+                    AND {% condition event_type %} event_types.event {% endcondition %}
+                ) AS selected_events
+              LEFT JOIN
+                UNNEST(event_properties) AS event_property
+              LEFT JOIN
+                UNNEST(value) AS event_property_value
+              WHERE
+                (
+                  event_property.key = "page"
+                  AND REGEXP_CONTAINS("""{% condition page %} __key__ {% endcondition %}""", r"__key__")
+                  AND {% condition page %} event_property_value.key {% endcondition %}
```

Comment:
Each incoming row is an event_property (e.g. `page`, `source`, etc.), and an associated value.

This specific check is for the event_property "page". If we filtered on it (second line), then we check that the filter is satisfied (third line).

If we did not filter on it, we don't want to include it here, but we want to pull it in from "all_event_properties".